### PR TITLE
Add default Token Endpoint Auth Method on `apps create` [CLI-115]

### PR DESCRIPTION
**Pull request by Widcket**
### Description

This PR adds a default value of `none` for the Token Endpoint Default Auth Method property on `apps create` when the Auth0 app a public client (SPA or Native).

Otherwise, the token exchange will fail with an `Oops... unauthorized` error.

Screen Shot 2021-03-31 at 13 14 05

Screen Shot 2021-03-31 at 13 20 12

### References

Documentation: https://auth0.com/docs/get-started/dashboard/application-settings

### Testing

- [ ] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [X] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used, if not `master`

### CODE REVIEW

**Review by aherzog-auth0 — APPROVED** [review]

File: internal/cli/apps.go [hunk 1]
```diff
@@ -668,6 +674,17 @@ func apiAuthMethodFor(v string) *string {
 	}
 }
 
+func apiDefaultAuthMethodFor(t string) *string {
+	switch apiTypeFor(strings.ToLower(t)) {
+	case appTypeNative:
+		fallthrough
+	case appTypeSPA:
+		return auth0.String("none")
```
Comment (cyx) [inline internal/cli/apps.go:682]: Just so I understand, we want `none` for both native/spa, correct?

Comment (Widcket) [reply internal/cli/apps.go:682]: Exactly, both SPA and native

File: internal/cli/apps.go [hunk 2]
```diff
@@ -666,6 +672,17 @@ func apiAuthMethodFor(v string) *string {
...
```
Comment (cyx) [inline internal/cli/apps.go:680]:
```suggestion
	case appTypeNative, appTypeSPA:
		return auth0.String("none")
```

**Review by cyx — APPROVED** [review]
Minor nit, otherwise 👍

File: internal/cli/apps.go (hunk 2 above)
Comment (Widcket) [reply internal/cli/apps.go:680]: Good catch, thanks!

**Pull request merged**